application/src/main/java/org/thingsboard/server/service/subscription/DefaultTbLocalSubscriptionService.java

```diff
@@ -266,18 +280,24 @@ public void cancelSubscription(String sessionId, int subscriptionId) {
     public void cancelAllSessionSubscriptions(String sessionId) {
         log.debug("[{}] Going to remove session subscriptions.", sessionId);
         List<SubscriptionModificationResult> results = new ArrayList<>();
-        subsLock.lock();
+        Lock subsLock = null;
         try {
             Map<Integer, TbSubscription<?>> sessionSubscriptions = subscriptionsBySessionId.remove(sessionId);
             if (sessionSubscriptions != null) {
                 for (TbSubscription<?> subscription : sessionSubscriptions.values()) {
```

Comment:
Isn't it too late to grab the lock? As I understand, sessionSubscriptions might be concurrently modified at this point (by `cancelSubscription`)

Reply:
We need to pass TenantId as an argument to `cancelAllSessionSubscriptions` and other similar methods that do not have the TenantId